Makefile

```diff
@@ -228,43 +214,21 @@ else
 endif
 
 compile_lib: config qemu/config-host.h-timestamp
-	cd qemu && $(MAKE) -j 4
-	$(MAKE) unicorn
+	$(MAKE) -C qemu -j 4
 
-unicorn: $(LIBRARY) $(ARCHIVE)
+unicorn: compile_lib $(LIBRARY) $(ARCHIVE)
 
-$(LIBRARY): $(UC_TARGET_OBJ) uc.o list.o
-ifeq ($(UNICORN_SHARED),yes)
-ifeq ($(V),0)
```

Comment:
if you look at autotools, this is specifically what it was designed to handle. instead of modifying makefiles, the platform dependencies are put into the autotools tools (https://www.gnu.org/software/automake/manual/automake.html#GNU-Build-System)